lms/validation/_lti_launch_params.py

```diff
@@ -164,3 +182,24 @@ class ConfigureAssignmentSchema(_CommonLTILaunchSchema):
     user_id = fields.Str(required=True)
     context_title = fields.Str(required=True)
     group_set = fields.Str(required=False, allow_none=True)
+    auto_grading_config = fields.Nested(
+        AutoGradingConfigSchema, required=False, allow_none=True
+    )
+
+    @pre_load
+    def _load_auto_grading_config(self, data, **_kwargs):
+        """Load auto grading config.
+
+        "form" location doesn't accept Nested fields we'll accept the value as json and deserilize it here.
```

Comment:
A bit of discussions around this here:

https://hypothes-is.slack.com/archives/C1M8NH76X/p1728557569201379

I reckon that while it is true that this is not very reusable, at least without some individualized change for each new nested field we want to support, I don't think we should try to make the "Nested fields on forms" better and try to move this API to a more REST/JSON style when possible.